core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -466,23 +531,13 @@ private[worker] class Worker(
         case _ =>
           logDebug(s"Driver $driverId changed state to $state")
       }
-      master ! DriverStateChanged(driverId, state, exception)
+      sendToMaster(driverStageChanged)
```

Comment:
This change was not strictly necessary, right? Maybe it used to have a self reference.